### PR TITLE
Feature/create raw data returns info

### PR DESCRIPTION
**Pull request by czkum**
When sending a `POST` request to the `RawDataService` in order to create a raw data, the endpoint with a minimum version of 1.7.0 responds a `RawDataInformationDto` object containing the key. This PR enables the `RawDataServiceRestClient` to respond this object. For endpoint versions younger than 1.7.0 a separate request is made internally which determines the result object so the user of the method can rely on having a response object in both version cases.

### CODE REVIEW

File: src/Api.Rest.Tests/HttpClient/RawData/RawDataServiceRestClientTest.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+			using var sut = new RawDataServiceRestClient( Uri, MaxUriLength );
+			var actual = await sut.CreateRawData( input, data );
+
+			Assert.That( actual.Key, Is.EqualTo( key ) );
+		}
+
+		private static void RegisterRawDataServiceVersionResponse( WebServer webServer, string version )
+		{
+			var supportedVersions = new StringBuilder();
+			supportedVersions.AppendLine( "{" );
```
Comment (czthiele) [inline src/Api.Rest.Tests/HttpClient/RawData/RawDataServiceRestClientTest.cs:108]: You could use the corresponding class InterfaceVersionRange and serialize it the same way you did in the methods below to avoid the manual json building.

File: src/Api.Rest/HttpClient/RawData/RawDataServiceRestClient.cs [hunk 2]
```diff
@@ -105,7 +105,33 @@ private async Task<RawDataInformationDto[]> ListRawDataByUuidList( RawDataEntity
...
+			var request = RequestBuilder.CreateWithAttachment( method, requestString, stream, info.MimeType, info.Size, info.MD5, info.FileName );
+
+			if ( featureMatrix.SupportsCreateRawDataResult )
+				return await _RestClient.Request<RawDataInformationDto>( request , cancellationToken );
+
+			await _RestClient.Request( request, cancellationToken );
+
+			var entity = info.Target.Entity;
+			var rawDataList = await ListRawData( entity, new[] { info.Target.Uuid }, null, cancellationToken );
+			return rawDataList.OrderByDescending( rawData => rawData.Key ?? 0 ).First();
```
Comment (czthiele) [inline src/Api.Rest/HttpClient/RawData/RawDataServiceRestClient.cs:131]: Maybe consider FirstOrDefault for the extreme edge case where the RawData has already been deleted between creation and refetch.

File: src/Api.Rest.Tests/HttpClient/Data/DataServiceRestClientTest.cs [hunk 3]
```diff
@@ -68,8 +68,8 @@ public async Task RequestSplit_OrderSpecified_OrderIsCorrect()
 					{ Attributes = new[] { new AttributeDto( 5, 3 ), new AttributeDto( 55, 20 ), new AttributeDto( 999, 3 ) } }
 			};
 
-			server.RegisterReponse( "/DataServiceRest/measurements?partUuids=%7B11111111-1111-1111-1111-111111111111%7D&order=5%20Asc%2C55%20Asc", JsonConvert.SerializeObject( firstMeasurementSet ) );
-			server.RegisterReponse( "/DataServiceRest/measurements?partUuids=%7B22222222-2222-2222-2222-222222222222%7D&order=5%20Asc%2C55%20Asc", JsonConvert.SerializeObject( secondMeasurementSet ) );
+			server.RegisterResponse( "/DataServiceRest/measurements?partUuids=%7B11111111-1111-1111-1111-111111111111%7D&order=5%20Asc%2C55%20Asc", JsonConvert.SerializeObject( firstMeasurementSet ) );
```
Comment (czthiele) [inline src/Api.Rest.Tests/HttpClient/Data/DataServiceRestClientTest.cs:71]: Thank you for noticing this typo 👍

File: src/Api.sln.DotSettings [hunk 4]
```diff
@@ -544,12 +544,14 @@ II.2.12 &lt;HandlesEvent /&gt;&#xD;
 	<s:String x:Key="/Default/CodeStyle/Naming/JavaScriptNaming/UserRules/=TS_005FPUBLIC_005FTYPE_005FFIELD/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="aaBb" /&gt;</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/JavaScriptNaming/UserRules/=TS_005FPUBLIC_005FTYPE_005FMETHOD/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="aaBb" /&gt;</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/JavaScriptNaming/UserRules/=TS_005FTYPE_005FPARAMETER/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="T" Suffix="" Style="AaBb" /&gt;</s:String>
+	<s:Boolean x:Key="/Default/Environment/Filtering/ExcludeCoverageFilters/=Cloud_002EData_003B_002A_003BZeiss_002EIMT_002EPiWeb_002ECloud_002EData_002EMigrations_002E_002A_003B_002A/@EntryIndexedValue">True</s:Boolean>
```
Comment (czthiele) [inline src/Api.sln.DotSettings:547]: Do you have an idea to which Rider setting this line belongs?

**Review by czthiele — APPROVED** [review]

File: src/Api.Rest.Tests/HttpClient/RawData/RawDataServiceRestClientTest.cs (hunk 1 above)
Comment (czkum) [reply src/Api.Rest.Tests/HttpClient/RawData/RawDataServiceRestClientTest.cs]: It's almost the same way as below ;) I had to use the VersionConverter to make it work.

File: src/Api.sln.DotSettings (hunk 4 above)
Comment (czkum) [reply src/Api.sln.DotSettings]: Not sure. But it wasn't my intention to change the settings. I've revoked it.

File: src/Api.Rest/HttpClient/RawData/RawDataServiceRestClient.cs (hunk 2 above)
Comment (czkum) [reply src/Api.Rest/HttpClient/RawData/RawDataServiceRestClient.cs:131]: Indeed, that's an edge case which should not happen in practice. In such a case I would not return null since the identification of the problem would then be handed over to the client. The information that the list doesn't contain any key would be lost.
Another edge case (and still very unlikely) is that the raw data has already been deleted and there is another key available. Then a wrong key would be returned.
I would not handle those cases since I cannot imagine how a raw data can be deleted immediately after creation.
